### ISSUE TITLE
Provide a quick way of updating submodules

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/civet#468]
Rattlesnake development quite frequently breaks its downstream applications. It will be nice if we can provide a gui in gitlab that we can use to selectively update submodules to a certain branch. For example in sabertooth, we can have a list of submodules and a list of branches user can click to choose. Once the selection is done, user can click submit to generate a MR. In this way, user does not have to check out a local repo and create the MR manually. It will be really handy! Tag @permcody @milljm

**Comment by YaqiWang** [comment]
Another nice thing about this is that an application developer does not have to have access to the downstream applications to send in such a MR.

**Comment by permcody** [comment]
Yes, this would be handy. However, I don't have the resources on my team to do this anytime in the near future.

One thing you can do to help mitigate breakage is to use deprecation. If there is any chance that you can keep the old and new APIs for a short period of time, you can let the system do updates and then remove the old later to avoid any immediate breakage. This method is highly recommended to save your sanity when it's possible.
